### PR TITLE
feature(js): datepicker will now show month and year selector

### LINKED ISSUE #5936: Display month and year menus in date input
Lets use a UI that has a dropdown for month and year:
http://jqueryui.com/datepicker/#dropdown-month-year

This makes it easy to select dates from a long time ago.

(Some people actually click the "previous" link hundreds of times to get to their birth year. And then they naturally complain about having to do it.)

### PR DESCRIPTION
**Pull request by jdalsem**
fixes #5936 

replaces #7832

### CODE REVIEW

**Comment by juho-jaakkola** [comment]
Great!

**Comment by jdalsem** [comment]
was a really difficult PR :)

File: js/lib/ui.js [hunk 1]
```diff
@@ -304,7 +304,9 @@ elgg.ui.initDatePicker = function() {
 				}
 			},
 			nextText: '&#xBB;',
-			prevText: '&#xAB;'
+			prevText: '&#xAB;',
+			changeMonth: true,
+			changeYear: true
```
Comment (ewinslow) [inline js/lib/ui.js]: Trailing commas are encouraged, btw

Comment (jdalsem) [inline js/lib/ui.js]: if we want es5 standards, we could also add it to scrutinizer or travis checks

Comment (jdalsem) [inline js/lib/ui.js]: i know we have no support for older version of IE, but still this could potentially break a lot of sites

Comment (mrclay) [inline js/lib/ui.js]: This would break IE7. I'm not concerned.

Comment (ewinslow) [inline js/lib/ui.js]: And IE8. Still not concerned

**Pull request merged**